pkg/server/router.go

```diff
@@ -290,6 +291,29 @@ func configHandler(server *config.Control, cfg *cmds.Server) http.Handler {
 	})
 }
 
+// configToCompareHandler exposes the config hash of the cluster. Joining control plane nodes will need to share the same config
+func configToCompareHandler(server *config.Control, cfg *cmds.Server) http.Handler {
+	configData, err := server.ControlQuorumArgs.Hash()
```

Comment:
This appears to be executed once to return the handler t hat will be executed everytime a call comes in. If that's the case, does it make sense to make the call to hash once and check the error each time the handler is called? If not, make the call once, check the error once and bail out. If the config can't be hashed, it's probably indicative of a larger problem.